File: cluster-check-go/service-accounts.go
package main

import (
	"context"
	"fmt"

	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
	"k8s.io/client-go/kubernetes"
)

func validateServiceAccounts(configCheck *Config, clientset *kubernetes.Clientset, sceCheck *bool, azureCheck *bool, omsCheck *bool) bool {
	fmt.Println(Separator)
	fmt.Println("Checking service accounts")
	fmt.Println(Separator)
	passed := true

	if *omsCheck {
		if !checkServiceAccounts(&configCheck.OMS.ServiceAccounts, clientset) {
			passed = false
		}
	}
	if *sceCheck {
		if !checkServiceAccounts(&configCheck.SCE.ServiceAccounts, clientset) {
			passed = false
		}
	}
	if *azureCheck {
		if !checkServiceAccounts(&configCheck.Azure.ServiceAccounts, clientset) {
			passed = false
		}
	}
	fmt.Print("\n")
	return passed
}

func checkServiceAccounts(serviceAccounts *[]NamespacedEntry, clientset *kubernetes.Clientset) bool {
	var passed = true
	for _, sa := range *serviceAccounts {
		status := ""
		color := ""
		_, err := clientset.CoreV1().ServiceAccounts(sa.Namespace).Get(context.TODO(), sa.Name, metav1.GetOptions{})
		if err != nil {
			status = "Not Found"
			color = Red
			passed = false
			checkError(err)
		} else {
			status = "Found"
			color = Green
		}
		fmt.Printf("%s/%s - %s%s %s\n", sa.Namespace, sa.Name, color, status, Reset)
	}
	return passed
}

File: cluster-check-py/serviceAccounts.py
from typing import List
import utils

from myTypes import Config, NamespacedEntry
from kubernetes.client import CoreV1Api
from kubernetes.client.rest import ApiException


def validate_service_accounts(config_check: Config, clientset, sce_check: bool, azure_check: bool, oms_check: bool):
    print("---------------------------------------------------------")
    print("Checking service accounts")
    print("---------------------------------------------------------")
    passed = True
    if oms_check:
        if not check_service_accounts(config_check.OMS.service_accounts, clientset):
            passed = False
    if sce_check:
        if not check_service_accounts(config_check.SCE.service_accounts, clientset):
            passed = False
    if azure_check:
        if not check_service_accounts(config_check.Azure.service_accounts, clientset):
            passed = False
    print()
    return passed


def check_service_accounts(service_accounts: List[NamespacedEntry], clientset):
    passed = True
    for sa in service_accounts:
        status = ""
        color = ""
        try:
            clientset.read_namespaced_service_account(sa.name, sa.namespace)
            status = "Found"
            color = utils.Color.GREEN.value
        except ApiException as e:
            status = "Not Found"
            color = utils.Color.RED.value
            passed = False
            utils.check_error(e)
        print(f"{sa.namespace}/{sa.name} - {color}{status}\033[0m")
    return passed